src/main/java/io/axoniq/axonserver/connector/command/impl/CommandChannelImpl.java

```diff
@@ -179,6 +179,7 @@ private synchronized void doCreateCommandStream() {
                        .reduce(CompletableFuture::allOf)
                        .map(cf -> cf.exceptionally(e -> {
                            logger.warn("An error occurred while registering command handlers", e);
+                           subscriptionsCompleted.set(false);
```

Comment:
Unfortunately, this won't make any difference. The ".thenRun()" operation will always be executed after this one.

The solution is to not map an exceptionally completed CompletableFuture, but instead just pass it on as such. Instead of thenRun(), use whenComplete(). It allows you to check whether there is an exception (and it thus completed exceptionally) or not. When completed exceptionally, log an error and set subscriptionsCompleted to false.

Reply:
You are right, I changed the implementation